iosxr_iso2vbox.py

```diff
@@ -65,38 +63,295 @@
+        child.expect("Enter secret:")
+        child.sendline("vagrant")
+        child.expect("Enter secret again:")
+        child.sendline("vagrant")
+        child.expect("Username:")
+        child.sendline("vagrant")
+        child.expect("Password:")
+        child.sendline("vagrant")
+        child.expect(prompt)
+
+        # Set term width
+        child.sendline("term width 300")
+        child.expect(prompt)
+
+        # ZTP causes some startup issues so disable it during box creation
+        child.sendline("ztp terminate noprompt")
+        child.expect(prompt)
+
+        # Determine if the image is a crypto/k9 image or not
+        # This will be used to determine whether to configure ssh or not
+        child.sendline("bash -c rpm -qa | grep k9sec")
+        output = child.expect('-k9sec')
+        if output == 0:
+            crypto = True
+            logger.debug("Crypto k9 image detected")
+        else:
+            crypto = False
+            logger.debug("Non crypto k9 image detected")
+
+        # Determine if the image has the MGBL package needed for gRPC
+        child.sendline("bash -c rpm -qa | grep mgbl")
+        output = child.expect('-mgbl')
+        if output == 0:
+            mgbl = True
+            logger.debug("MGBL package detected")
+        else:
+            mgbl = False
+            logger.debug("MGBL package not detected")
+
+        # Wait for a management interface to be available
+        xr_cli_wait_for_output('sh run | inc MgmtEth', 'interface MgmtEth')
+
+        child.sendline("conf t")
+        child.expect("ios.config.*#", 10)
+
+        # Enable telnet
+        child.sendline("telnet vrf default ipv4 server max-servers 10")
+        child.expect("config")
+
+        # Bring up dhcp on MGMT for vagrant access
+        child.sendline("interface MgmtEth0/RP0/CPU0/0")
+        child.sendline(" ipv4 address dhcp")
+        child.sendline(" no shutdown")
+        child.expect("config-if")
+
+        # TPA source update
+        child.sendline("tpa address-family ipv4 update-source MgmtEth0/RP0/CPU0/0")
+        child.expect("config")
+
+        child.sendline("router static address-family ipv4 unicast 0.0.0.0/0 MgmtEth0/RP0/CPU0/0 10.0.2.2")
+        child.expect("config")
+
+        # Configure ssh if a k9/crypto image
+        if crypto:
+            child.sendline("ssh server v2")
+            child.expect("config")
+            child.sendline("ssh server vrf default")
+            child.expect("config")
+
+        # Configure gRPC protocol if MGBL package is available
+        if mgbl:
+            child.sendline("grpc")
+            child.sendline(" port 57777")
+            child.expect("config-grpc")
+
+        # Commit changes and end
+        child.sendline("commit")
+        child.expect("config")
+
+        child.sendline("end")
+        child.expect(prompt)
+
+        # Spin waiting for an ip address to be associated with the interface
+        while True:
```

Comment:
This should be using xr_cli_wait_for_output() - not sure how I left that out but adding it back in and retesting.